src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ThrottlingRerouteSpec.groovy

```diff
@@ -152,20 +157,22 @@ class ThrottlingRerouteSpec extends HealthCheckSpecification {
         then: "Right until hard timeout should run out no flow reroutes happen"
         //check until 80% of hard timeout runs out
         while (System.currentTimeMillis() < rerouteTriggersStart.time + rerouteHardTimeout * 1000 * 0.8) {
-            flowPaths.each { flowPath ->
-                assert northboundV2.getFlowStatus(flowPath.id).status == FlowState.UP &&
-                        northbound.getFlowPath(flowPath.id) == flowPath
+            flows.each { flow ->
+                def initialFlowPath = flowPaths.find { it.flowPath.flowId == flow.flowId }
+                assert flow.retrieveFlowStatus().status == FlowState.UP &&
+                        flow.retrieveAllEntityPaths().getPathNodes() == initialFlowPath
             }
         }
 
-        and: "Flows should start to reroute after hard timeout, eventhough reroutes are still being triggered"
+        and: "Flows should start to reroute after hard timeout, even though reroutes are still being triggered"
         rerouteTriggers.any { it.alive }
         def flowPathsClone = flowPaths.collect()
         Wrappers.wait(untilHardTimeoutEnds() + WAIT_OFFSET) {
             flowPathsClone.removeAll { flowPath ->
-                (northboundV2.getFlowStatus(flowPath.id).status == FlowState.DOWN && northbound
-                        .getFlowHistory(flowPath.id).last().payload.find { it.action == REROUTE_FAIL }) ||
-                        northbound.getFlowPath(flowPath.id) != flowPath
+                def flow = flows.find { it.flowId == flowPath.flowPath.flowId }
+                def lastFlowAction = flow.retrieveFlowHistory().getEntriesByType(FlowActionType.REROUTE_FAILED).last().toString()
+                (flow.retrieveFlowStatus().status == FlowState.DOWN && lastFlowAction == REROUTE_FAIL)
+                        || flow.retrieveAllEntityPaths().getPathNodes() != flowPath
```

Comment:
Could you please replace REROUTE_FAIL with FlowActionType.REROUTE_FAILED   
old version:
`northbound .getFlowHistory(flowPath.id).last().payload.find { it.action == REROUTE_FAIL }) `-the  last payload of the last event 
new version
`def lastFlowAction = flow.retrieveFlowHistory().getEntriesByType(FlowActionType.REROUTE_FAILED).last().toString()` - it is just last reroute event

Reply:
Fixed